test/src/test/java/org/corfudb/integration/WorkflowIT.java

```diff
@@ -360,4 +363,182 @@ public void addNodeWithTrim() throws Exception {
+                })
+                .setStreamName("test")
+                .open();
+
+        // Do initial write on the table
+        corfuRuntime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
+        table.put(initKey, String.valueOf(initKey));
+        corfuRuntime.getObjectsView().TXEnd();
+
+        initKey++;
+
+        CountDownLatch countDownLatch1 = new CountDownLatch(1);
+        CountDownLatch countDownLatch2 = new CountDownLatch(1);
+        CountDownLatch countDownLatch3 = new CountDownLatch(1);
+
+        Thread t = new Thread(() -> {
+            corfuRuntime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
+
+            table.put(numElements, String.valueOf(numElements));
+                    countDownLatch1.countDown();
+            try {
+                countDownLatch2.await();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            // TODO(Anny): when debugging found that on thnis access the VLO is reset, not sure if this is
+            // efficient or there is a bug, further look into this...
+            table.get(0);
+            corfuRuntime.getObjectsView().TXEnd();
+            countDownLatch3.countDown();
+        });
+        t.start();
+
+        countDownLatch1.await();
+
+        for (int i = initKey; i < numElements - 2; i++) {
+            corfuRuntime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
+            table.put(i, String.valueOf(i));
+            corfuRuntime.getObjectsView().TXEnd();
+        }
+
+        MultiCheckpointWriter mcw = new MultiCheckpointWriter();
+        mcw.addMap(table);
+        Token prefixTrim = mcw.appendCheckpoints(corfuRuntime, "author");
+
+        for (int i = numElements - 2; i < numElements; i++) {
+            corfuRuntime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
+            table.put(i, String.valueOf(i));
+            corfuRuntime.getObjectsView().TXEnd();
+        }
+
+        countDownLatch2.countDown();
+        countDownLatch3.await();
+
+        corfuRuntime.getAddressSpaceView().prefixTrim(prefixTrim);
+        corfuRuntime.getGarbageCollector().runRuntimeGC();
+
+        // Assert that after running GC we are not missing data.
+        assertThat(table).hasSize(numElements + 1);
+        for(int i = 0; i <= numElements; i++) {
+            assertThat(table.get(i)).isEqualTo(String.valueOf(i));
+        }
+
+        t.join();
+
+        boolean exceptionCaught = false;
+        try {
+            final int snapshotTxAddress = 4;
+            corfuRuntime.getObjectsView().TXBuild().type(TransactionType.SNAPSHOT)
+                    .snapshot(new Token(0, snapshotTxAddress))
+                    .build().begin();
+            table.get(0);
+            corfuRuntime.getObjectsView().TXEnd();
```

Comment:
You can put a `fail()` here and get rid of the exceptionCaught assertion.
Or you can look at  `assertThrownBy` and assert that it throws a transaction aborted exception with cause trim.

Reply:
Done.